projects/kit/components/input-range/input-range.component.ts

```diff
@@ -0,0 +1,225 @@
+        tuiAsControl(TuiInputRangeComponent),
+        tuiSliderOptionsProvider({trackColor: 'transparent'}),
+        tuiFallbackValueProvider([0, 0]),
+        tuiValueTransformerFrom(TUI_INPUT_RANGE_OPTIONS),
+    ],
+    host: {
+        new: '', // TODO(v5): remove after deletion of legacy control
+        // TODO: use css :host:has(tui-textfield[data-size]) after browser bump
+        '[attr.data-size]': 'size()',
+    },
+})
+export class TuiInputRangeComponent
+    extends TuiControl<readonly [number, number]>
+    implements AfterViewInit
+{
+    @ViewChildren(TuiInputNumberDirective, {read: ElementRef})
+    private readonly inputNumberRefs: QueryList<ElementRef<HTMLInputElement>> =
+        EMPTY_QUERY;
+
+    @ViewChild(TuiRange, {read: ElementRef})
+    private readonly rangeRef?: ElementRef<HTMLElement>;
+
+    @ViewChild(TuiRange)
+    private readonly range?: TuiRange;
+
+    private readonly isMobile = inject(TUI_IS_MOBILE);
+    private readonly options = inject(TUI_INPUT_RANGE_OPTIONS);
+    private readonly quantum = signal(0);
+    private readonly quantumTransformer = computed((quantum = this.quantum()) =>
+        quantum
+            ? new TuiQuantumValueTransformerBase(quantum)
+            : TUI_IDENTITY_VALUE_TRANSFORMER,
+    );
+
+    protected readonly size = inject(TUI_TEXTFIELD_OPTIONS).size;
+    protected textfieldValueStart = this.value()[0];
+    protected textfieldValueEnd = this.value()[1];
+    protected lastActiveSide: 'end' | 'start' = 'start';
+
+    @Input()
+    public min = this.options.min;
+
+    @Input()
+    public max = this.options.max;
+
+    @Input()
+    public step = this.options.step;
+
+    @Input()
+    public segments = this.options.segments;
+
+    @Input()
+    public keySteps: TuiKeySteps | null = this.options.keySteps;
+
+    @Input()
+    public content: readonly [
+        PolymorpheusContent<TuiContext<number>>,
+        PolymorpheusContent<TuiContext<number>>,
+    ] = this.options.content;
+
+    @Input({transform: (x: readonly [string, string] | null) => x ?? ['', '']})
+    public prefix: readonly [string, string] = this.options.prefix;
+
+    @Input({transform: (x: readonly [string, string] | null) => x ?? ['', '']})
+    public postfix: readonly [string, string] = this.options.postfix;
+
+    // TODO(v5): use signal inputs
+    @Input('quantum')
+    public set quantumSetter(x: number) {
+        this.quantum.set(x);
+    }
+
+    public override writeValue(value: [number, number]): void {
+        super.writeValue(value);
+        this.setTextfieldValues(this.value());
+    }
+
+    public ngAfterViewInit(): void {
+        if (this.range) {
+            this.range.legacyMode = false; // TODO(v5): remove backward compatibility
+        }
+    }
+
+    protected get hideStartContent(): boolean {
+        return (
+            !this.content[0] ||
+            tuiIsNativeFocused(this.textfieldStart) ||
+            (tuiIsNativeFocusedIn(this.rangeRef?.nativeElement) &&
```

Comment:
Is focus ever in the range component? I failed to reproduce it by mouse or keyboard

Reply:
You are right. Probably, some legacy rudiment. I'll remove it